target_chains/ton/contracts/contracts/Pyth.fc

```diff
@@ -17,7 +17,7 @@ cell store_price(int price, int conf, int expo, int publish_time) {
     .end_cell();
 }
 
-slice read_and_verify_header(slice data) {
+slice read_and_verify_header(slice data) impure {
```

Comment:
why did it turn impure? because we use a global variable? strange (because it's const)

Reply:
they should be marked impure because it can throw exceptions using `throw_unless` and `throw_if` according to [FunC docs here](https://docs.ton.org/v3/documentation/smart-contracts/func/docs/functions#impure-specifier) otherwise it might be incorrectly optimized away